comp/dogstatsd/server/rc.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"encoding/json"
+	"slices"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+)
+
+type statsdBlocklistUpdate struct {
+	BlockedMetrics blockedMetrics `json:"blocked_metrics"`
+}
+
+type blockedMetrics struct {
+	ByName byName `json:"by_name"`
+}
+
+type byName struct {
+	ConfigurationID int           `json:"configuration_id"`
+	Metrics         []metricEntry `json:"values"`
+}
+
+type metricEntry struct {
+	Name string `json:"metric_name"`
+}
+
+func (s *server) onBlocklistUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debug("onBlocklistUpdateCallback received updates:", len(updates))
+
+	if len(updates) == 0 {
+		return
+	}
+
+	var blocklistUpdates []blockedMetrics
+
+	// unmarshal all the configurations received from
+	// the RC platform
+	// ------------------------
```

Comment:
For what it's worth: I noticed you've added a number of comments in this PR using this format.

While I personally love code comments, I think this format -- in particular, the last line always being `-----------------`, is unique amongst all existing code comments and feels a little out of place.

I would suggest just removing the separator lines, but everything else is fine.